app/src/test/java/com/davidtomas/taskyapp/features/auth/presentation/login/LoginViewModelTest.kt

```diff
@@ -0,0 +1,160 @@
+                        UiText.StringResource(
+                            R.string.login_success
+                        )
+                    )
+                )
+            }
+        }
+
+    @Test
+    fun `when login button is clicked and fields are incorrectly filled, login returns error`() =
+        runTest {
+            var eventCollected: LoginUiEvent?
+            val newEmail = "newEmail"
+            val newPassword = "wrongPassword"
+            coEvery { authRepository.login(any()) } returns Result.Error(DataError.Network.BAD_REQUEST)
+            loginViewModel.onAction(LoginAction.OnEmailChanged(newEmail))
+            loginViewModel.onAction(LoginAction.OnPasswordChanged(newPassword))
+
+            loginViewModel.uiEvent.test {
+                loginViewModel.onAction(LoginAction.OnLoginButtonClick)
+                eventCollected = awaitItem()
+                assertThat(eventCollected).isEqualTo(
+                    LoginUiEvent.ShowSnackBar(
+                        UiText.StringResource(
+                            R.string.error_bad_request
+                        )
+                    )
+                )
+            }
+        }
+
+    @Test
+    fun `when login button is clicked and email is empty, informative snackbar is shown`() =
+        runTest {
+            var eventCollected: LoginUiEvent?
+            val newEmail = String.EMPTY_STRING
+            val newPassword = "wrongPassword"
+            coEvery { authRepository.login(any()) } returns Result.Error(DataError.Network.BAD_REQUEST)
+            loginViewModel.onAction(LoginAction.OnEmailChanged(newEmail))
+            loginViewModel.onAction(LoginAction.OnPasswordChanged(newPassword))
+
+            loginViewModel.uiEvent.test {
+                loginViewModel.onAction(LoginAction.OnLoginButtonClick)
+                eventCollected = awaitItem()
+                assertThat(eventCollected).isEqualTo(
+                    LoginUiEvent.ShowSnackBar(
+                        UiText.StringResource(
+                            R.string.error_fields_must_be_filled
+                        )
+                    )
+                )
+            }
+        }
+
+    @Test
+    fun `when login button is clicked and password is empty, informative snackbar is shown`() =
+        runTest {
+            var eventCollected1: LoginUiEvent?
+            var eventCollected2: LoginUiEvent?
+            val newEmail = "newEmail"
+            val newPassword = String.EMPTY_STRING
+            coEvery { authRepository.login(any()) } returns Result.Error(DataError.Network.BAD_REQUEST)
+            loginViewModel.onAction(LoginAction.OnEmailChanged(newEmail))
+            loginViewModel.onAction(LoginAction.OnPasswordChanged(newPassword))
+
+            loginViewModel.uiEvent.test {
+                loginViewModel.onAction(LoginAction.OnLoginButtonClick)
+                eventCollected1 = awaitItem()
+                eventCollected2 = awaitItem()
```

Comment:
Could be that you're missing an `advanceUntilIdle` in between

Reply:
Yeah, Working!! Thanks